Language Overview broken into separate files & without numeric headings (#4576)
brimdata/zed: docs/language/README.md

The language documents:
* provide an [overview](overview.md) of the Zed language,
* describe Zed's [dataflow model](dataflow-model.md),

> * show the syntax of [statements](statements.md) that define constants, functions, and named types,
> * explain how Zed [data types](data-types.md) are used,

Reviewer: ```suggestion
* explain Zed's [data types](data-types.md),
* show the syntax of [statements](statements.md) that define constants, functions, and named types,

```

> * describe [expression](expressions.md) syntax, including for [search](search-expressions.md),

Reviewer: ```suggestion
* describe the syntax of [expressions](expressions.md) and [search expressions](search-expressions.md),
```

> * explain how [lateral queries](lateral-subqueries.md) are used,

Reviewer: ```suggestion
* explain [lateral subqueries](lateral-subqueries.md),
```

> * describe the the concepts of [shaping and type fusion](shaping.md), and

Author: It felt awkward to write such a long extension of this narrative reveal of hyperlinks, so if folks wanna go another direction here, I'm open to that.

> * describe the the concepts of [shaping and type fusion](shaping.md), and

Reviewer: ```suggestion
* describe [shaping and type fusion](shaping.md), and
```
